src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -724,6 +725,24 @@ public IEnumerable<SyntaxNode> AncestorsAndSelf(bool ascendOutOfTrivia = true)
             return null;
         }
 
+        /// <summary>
+        /// Gets the first node of type TNode that matches the predicate.
+        /// </summary>
+        [SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Required for consistent API usage patterns.")]
+        public TNode? FirstAncestorOrSelf<TNode, TArg>(Func<TNode, TArg, bool> predicate, TArg argument, bool ascendOutOfTrivia = true)
```

Comment:
It would be nice to have an analyzer to find these kind of overloads and suggest to use it if the passed lambda is capering something and possibly auto fix! (using tuples for >1 captures)

Reply:
We could certainly consider having an analyzer like this in roslyn-analyzers.